src/sentry/issues/endpoints/organization_group_search_view_starred_order.py

```diff
@@ -0,0 +1,99 @@
+from django.db import IntegrityError, router, transaction
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.models.groupsearchview import GroupSearchView, GroupSearchViewVisibility
+from sentry.models.groupsearchviewstarred import GroupSearchViewStarred
+from sentry.models.organization import Organization
+
+
+class MemberPermission(OrganizationPermission):
+    scope_map = {
+        "PUT": ["member:read", "member:write"],
+    }
+
+
+class GroupSearchViewStarredOrderSerializer(serializers.Serializer):
+    view_ids = serializers.ListField(child=serializers.IntegerField(), required=True, min_length=0)
+
+    def validate_view_ids(self, view_ids):
+        if len(view_ids) != len(set(view_ids)):
+            raise serializers.ValidationError("Single view cannot take up multiple positions")
+
+        try:
+            gsvs = GroupSearchView.objects.filter(
+                organization=self.context["organization"], id__in=view_ids
+            )
```

Comment:
could use `.values(...)` here to make the query a bit more efficient.

it is also probably good to do validation here that all the gsvs exist. that is, check the length of view_ids against the returned query.

Reply:
oh i guess it not existing is valid input, since we create it below